src/webgpu/api/operation/command_buffer/copyTextureToTexture.spec.ts

```diff
@@ -325,5 +197,284 @@ g.test('t2t_non_compressed_color_formats')
     }
 
     // Verify the content of the whole subresouce of dstTexture at dstCopyLevel (in dstBuffer) is expected.
-    t.expectContents(dstBuffer, expectedUint8DataWithPadding);
+    this.expectContents(dstBuffer, expectedUint8DataWithPadding);
+  }
+
+  static kCopyBoxOffsetsForWholeDepth = [
+    // From (0, 0) of src to (0, 0) of dst.
+    {
+      srcOffset: { x: 0, y: 0, z: 0 },
+      dstOffset: { x: 0, y: 0, z: 0 },
+      copyExtent: { width: 0, height: 0, depth: 0 },
+    },
+    // From (0, 0) of src to (1, 0) of dst.
+    {
+      srcOffset: { x: 0, y: 0, z: 0 },
+      dstOffset: { x: 1, y: 0, z: 0 },
+      copyExtent: { width: 0, height: 0, depth: 0 },
+    },
+    // From (0, 0) of src to (0, 1) of dst.
+    {
+      srcOffset: { x: 0, y: 0, z: 0 },
+      dstOffset: { x: 0, y: 1, z: 0 },
+      copyExtent: { width: 0, height: 0, depth: 0 },
+    },
+    // From (1, 0) of src to (0, 0) of dst.
+    {
+      srcOffset: { x: 1, y: 0, z: 0 },
+      dstOffset: { x: 0, y: 0, z: 0 },
+      copyExtent: { width: 0, height: 0, depth: 0 },
+    },
+    // From (0, 1) of src to (0, 0) of dst.
+    {
+      srcOffset: { x: 0, y: 1, z: 0 },
+      dstOffset: { x: 0, y: 0, z: 0 },
+      copyExtent: { width: 0, height: 0, depth: 0 },
+    },
+    // From (1, 0) of src to (0, 0) of dst, and the copy extent will not cover the last
+    // column of both source and destination texture.
+    {
+      srcOffset: { x: 1, y: 0, z: 0 },
+      dstOffset: { x: 0, y: 0, z: 0 },
+      copyExtent: { width: -1, height: 0, depth: 0 },
+    },
+    // From (0, 1) of src to (0, 0) of dst, and the copy extent will not cover the last
+    // row of both source and destination texture.
+    {
+      srcOffset: { x: 0, y: 1, z: 0 },
+      dstOffset: { x: 0, y: 0, z: 0 },
+      copyExtent: { width: 0, height: -1, depth: 0 },
+    },
+  ];
+}
+
+export const g = makeTestGroup(F);
+
+g.test('color_textures_non_compressed_non_array')
+  .desc(
+    `
+  Validate the correctness of the copy by filling the srcTexture with testable data and any non-compressed color format that is
+  supported by WebGPU, doing CopyTextureToTexture() copy, and verifying the content of the whole dstTexture
+  TODO(jiawei.shao@intel.com): support all WebGPU texture formats
```

Comment:
nit (sorry, missed this before): Having the TODO here implies we'll add it to this test. If you think we'll add it to this test, update the test name to not include "non_compressed"; if you think it'll be a separate test, move this TODO into the file description. Same below.